more_itertools/more.pyi

```diff
@@ -672,3 +672,6 @@ def constrained_batches(
 ) -> Iterator[tuple[_T]]: ...
 def gray_product(*iterables: Iterable[_T]) -> Iterator[tuple[_T, ...]]: ...
 def partial_product(*iterables: Iterable[_T]) -> Iterator[tuple[_T, ...]]: ...
+def full_outer_join(
+    *iterables: Iterable[_T], keyfunc: Callable[[_T], _U]
+) -> Iterator[tuple[_U, tuple[list[_T]]]]: ...
```

Comment:
Not sure if I did this right. The inner tuple has the same arity as *iterables, how do you express that?